shell/renderer/api/electron_api_web_frame.cc

```diff
@@ -349,555 +349,640 @@ class SpellCheckerHolder final : public content::RenderFrameObserver {
+    return suggestions;
+  }
+#endif
 
-  blink::WebFrame* frame = render_frame->GetWebFrame()->FirstChild();
-  if (frame && frame->IsWebLocalFrame())
-    return frame->ToWebLocalFrame()->MainWorldScriptContext()->Global();
-  else
-    return v8::Null(isolate);
-}
+  void ClearCache(v8::Isolate* isolate) {
+    isolate->IdleNotificationDeadline(0.5);
+    blink::WebCache::Clear();
+    base::MemoryPressureListener::NotifyMemoryPressure(
+        base::MemoryPressureListener::MEMORY_PRESSURE_LEVEL_CRITICAL);
+  }
 
-v8::Local<v8::Value> GetNextSibling(v8::Isolate* isolate,
-                                    v8::Local<v8::Value> window) {
-  auto* render_frame = GetRenderFrame(window);
-  if (!render_frame)
-    return v8::Null(isolate);
+  v8::Local<v8::Value> FindFrameByRoutingId(v8::Isolate* isolate,
+                                            int routing_id) {
+    content::RenderFrame* render_frame =
+        content::RenderFrame::FromRoutingID(routing_id);
+    if (render_frame)
+      return WebFrameRenderer::Create(isolate, render_frame).ToV8();
+    else
+      return v8::Null(isolate);
+  }
 
-  blink::WebFrame* frame = render_frame->GetWebFrame()->NextSibling();
-  if (frame && frame->IsWebLocalFrame())
-    return frame->ToWebLocalFrame()->MainWorldScriptContext()->Global();
-  else
-    return v8::Null(isolate);
-}
+  v8::Local<v8::Value> GetOpener(gin_helper::ErrorThrower thrower,
+                                 v8::Isolate* isolate) {
+    content::RenderFrame* render_frame;
+    if (!MaybeGetRenderFrame(thrower, "opener", &render_frame))
+      return v8::Null(isolate);
+
+    blink::WebFrame* frame = render_frame->GetWebFrame()->Opener();
+    if (frame && frame->IsWebLocalFrame())
+      return WebFrameRenderer::Create(
+                 isolate,
+                 content::RenderFrame::FromWebFrame(frame->ToWebLocalFrame()))
+          .ToV8();
+    else
+      return v8::Null(isolate);
+  }
 
-v8::Local<v8::Value> GetFrameForSelector(v8::Isolate* isolate,
-                                         v8::Local<v8::Value> window,
-                                         const std::string& selector) {
-  blink::WebElement element =
-      GetRenderFrame(window)->GetWebFrame()->GetDocument().QuerySelector(
-          blink::WebString::FromUTF8(selector));
-  if (element.IsNull())  // not found
-    return v8::Null(isolate);
+  // Don't name it as GetParent, Windows has API with same name.
+  v8::Local<v8::Value> GetFrameParent(gin_helper::ErrorThrower thrower,
+                                      v8::Isolate* isolate) {
+    content::RenderFrame* render_frame;
+    if (!MaybeGetRenderFrame(thrower, "parent", &render_frame))
+      return v8::Null(isolate);
+
+    blink::WebFrame* frame = render_frame->GetWebFrame()->Parent();
+    if (frame && frame->IsWebLocalFrame())
+      return WebFrameRenderer::Create(
+                 isolate,
+                 content::RenderFrame::FromWebFrame(frame->ToWebLocalFrame()))
```

Comment:
What does this change do..?

Reply:
This specifically, changes `GetFrameParent` to return a new `WebFrameRenderer` instance for the given RenderFrame instead of just returning that frames global object.

Reply:
Ah, I see, we used to do that wrapping on the JS side. 👍